### PR TITLE
Removed jsonlint-cli and gulp-jsonlint. Replaced with eslint-plugin-json

### LINKED ISSUE #14621: tests: Move from jsonlint-cli to eslint-plugin-json
We currently use a jsonlint-cli to lint our json files, but this hasn't worked that well with Windows machines. 

There is now an eslint plugin module for linting json files. We should move to using this plugin to lint everything. 

This should involve adding the plugin to both package.json dev dependencies and to the plugins section of .eslintrc. You will also need to update the npm scripts that run lint tasks.

This should greatly simplify our linting story.

### PR DESCRIPTION
**Pull request by Ethan-Arrowood**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [x] Closes currently open issue (replace XXXX with an issue no): Closes #14621

#### Description

Replaced old jsonlint-cli with eslint-plugin-json.
In package.json I uninstalled the jsonlint-cli and gulp-jsonlint modules and dependencies.
In .eslintrc I updated the plugins to include json, and the settings to no longer ignore .json files.
In gulpfile.js I changed the gulp task to run eslint instead of jsonlint. 

Additionally there was some strange lint issue with 2 unknown characters in a Russian translation in some challenge file.

### CODE REVIEW

File: seed/challenges/01-responsive-web-design/basic-css.json [hunk 1]
```diff
@@ -2924,7 +2924,7 @@
           "description": [
             "Мы только что доказали, что браузеры читают CSS сверху-вниз. Это значит, что в случае конфликта значений будет установлено то, которое было объявлено в последнюю очередь.",
             "Но мы ещё не закончилил. Есть и другие способы переопределения CSS. Помните атрибуты id?",
-            "Давайте переопределим ваши классы <code>pink-text</code> и <code>blue-text</code>, и сделаем ваш элемент <code>h1</code> оранжевым, назначив элементу <code>h1</code> зна��ение атрибута <code>id</code> и его последующей стилизацией.",
+            "Давайте переопределим ваши классы <code>pink-text</code> и <code>blue-text</code>, и сделаем ваш элемент <code>h1</code> оранжевым, назначив элементу <code>h1</code> знаение атрибута <code>id</code> и его последующей стилизацией.",
```
Comment (BerkeleyTrue) [inline seed/challenges/01-responsive-web-design/basic-css.json]: You have an unnecessary change here. Please revert and amend you commit

File: package.json [hunk 2]
```diff
@@ -10,13 +10,9 @@
...
     "start-production": "node pm2Start",
     "lint": "npm run lint-js && npm run lint-json",
-    "lint-challenges": "jsonlint-cli seed/challenges/**/*.json",
-    "lint-server": "jsonlint-cli server/*.json",
-    "lint-resources": "jsonlint-cli server/resources/*.json",
-    "lint-utils": "jsonlint-cli server/utils/*.json",
     "prelint-js": "npm run create-rev",
     "lint-js": "eslint --ext=.js,.jsx gulpfile.js server/ common/ config/ client/",
-    "lint-json": "npm run lint-server && npm run lint-challenges && npm run lint-resources && npm run lint-utils",
+    "lint-json": "eslint --ext=.json seed/challenges/ server/",
```
Comment (BerkeleyTrue) [inline package.json]: fix: Merge these two commands so that eslint is only run once

**Review by BerkeleyTrue — CHANGES_REQUESTED** [review]

File: seed/challenges/01-responsive-web-design/basic-css.json (hunk 1 above)
Comment (Ethan-Arrowood) [inline seed/challenges/01-responsive-web-design/basic-css.json]: Tests wouldn't pass without changing this :(

File: package.json (hunk 2 above)
Comment (Ethan-Arrowood) [inline package.json]: Will do

File: seed/challenges/01-responsive-web-design/basic-css.json (hunk 1 above)
Comment (BerkeleyTrue) [inline seed/challenges/01-responsive-web-design/basic-css.json]: Is it failing on lint?

File: package.json (hunk 2 above)
Comment (Ethan-Arrowood) [inline package.json]: So i've come up with three solutions:
1. Eslint is ran twice, but condense into one long command.
`"lint": "eslint --ext=.js,.jsx gulpfile.js server/ common/ config/ client/ && eslint --ext=.json seed/challenges/ server/", `
2. Eslint is only ran once; however, this lints additional .js/.json files 
`"lint": "eslint --ext=.js,.jsx,.json gulpfile.js server/ common/ config/ client/ seed/challenges/ ",`
3. Appears to do the same as command 2
`"lint": "eslint --ext=.js,.jsx gulpfile.js server/ common/ config/ client/ --ext=.json seed/challenges/ server/", `

For commands 2 and 3 the following lint errors are produced:
```
/Users/arrowoode/Documents/FreeCodeCamp/freeCodeCamp/common/app/ns.json
  1:6  error  Missing semicolon  bad-json

/Users/arrowoode/Documents/FreeCodeCamp/freeCodeCamp/common/app/routes/challenges/ns.json
  1:13  error  Missing semicolon  bad-json

/Users/arrowoode/Documents/FreeCodeCamp/freeCodeCamp/common/app/routes/challenges/views/classic/ns.json
  1:10  error  Missing semicolon  bad-json

/Users/arrowoode/Documents/FreeCodeCamp/freeCodeCamp/common/app/routes/challenges/views/step/ns.json
  1:7  error  Missing semicolon  bad-json

✖ 4 problems (4 errors, 0 warnings)
```
If you'd like to use command 2, but ignore the files causing issues we can add `common/.json` to the .eslintignore file

Comment (BerkeleyTrue) [inline package.json]: I see.

We should go with option 2 so that all json files are linted.

The error is troubling as it assumes the previous spec of JSON where JSON must be an object or an array. Current spec states that JSON must be a value. 

After taking a look under the hood, it looks like eslint-plugin-json uses jshint under the hood. It may be that jshint uses an outdated spec to lint json. This maybe a no go for this plugin. Let me see if I can put something together.

File: seed/challenges/01-responsive-web-design/basic-css.json (hunk 1 above)
Comment (Ethan-Arrowood) [inline seed/challenges/01-responsive-web-design/basic-css.json]: Yes it is

**Comment by BerkeleyTrue** [comment]
@Ethan-Arrowood Thanks but the fact that eslint-plugin-json uses an out of date spec means we can't use this package. I may use my free time to clone the package and use jsonlint under the hood. If I do, then I'll post in the issue and you can just re-use this PR. 

Closing for now.
